v0/destinations/hs/util.js

```diff
@@ -1,12 +1,126 @@
-const { send } = require("../../../adapters/network");
+const get = require("get-value");
+const { httpGET, httpPOST } = require("../../../adapters/network");
+const {
+  processAxiosResponse
+} = require("../../../adapters/utils/networkUtils");
+const {
+  getFieldValueFromMessage,
+  constructPayload,
+  CustomError,
+  isEmpty,
+  getHashFromArray
+} = require("../../util");
+const {
+  CONTACT_PROPERTY_MAP_ENDPOINT,
+  IDENTIFY_CRM_SEARCH_CONTACT
+} = require("./config");
 
-const getAllContactProperties = async endpoint => {
-  const requestOptions = {
-    url: endpoint,
-    method: "get"
-  };
-  const res = await send(requestOptions);
-  return res;
+const formatKey = key => {
+  // lowercase and replace spaces and . with _
+  let modifiedKey = key.toLowerCase();
+  modifiedKey = modifiedKey.replace(/\s+/g, "_");
+  modifiedKey = modifiedKey.replace(/\./g, "_");
+  return modifiedKey;
+};
+
+const getTraits = message => {
+  // get from traits or properties
+  let traits = getFieldValueFromMessage(message, "traits");
+  if (!traits || !Object.keys(traits).length) {
+    traits = message.properties;
+  }
+  return traits;
+};
+
+const getProperties = async destination => {
+  let hubSpotPropertyMap = {};
+  let res;
+  const { Config } = destination;
+
+  // select API authorization type
+  if (Config.authorizationType === "newPrivateAppApi") {
+    // Private Apps
+    const requestOptions = {
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${Config.accessToken}`
+      }
+    };
+    res = await httpGET(CONTACT_PROPERTY_MAP_ENDPOINT, requestOptions);
+  } else {
+    // API Key (hapikey)
+    const url = `${CONTACT_PROPERTY_MAP_ENDPOINT}?hapikey=${Config.apiKey}`;
+    res = await httpGET(url);
```

Comment:
after getting a result from `http` wrapper functions use `processAxiosResponse` to process the response to standard format